lua_modules/dht22/dht22.lua

```diff
@@ -0,0 +1,101 @@
+-- ***************************************************************************
+-- DHT22 module for ESP8266 with nodeMCU
+--
+-- Written by Javier Yanez 
+-- but based on a script of Pigs Fly from ESP8266.com forum
+--
+-- MIT license, http://opensource.org/licenses/MIT
+-- ***************************************************************************
+
+local moduleName = ...
+local M = {}
+_G[moduleName] = M
+
+local humidity
+local temperature
+local checksum
+local checksumTest
+
+function M.read(pin)
+  humidity = 0
+  temperature = 0
+  checksum = 0
+
+  -- Use Markus Gritsch trick to speed up read/write on GPIO
+  gpio_read = gpio.read
```

Comment:
It's just due to `gpio.read()` requires: access to global table, access to its field, call vs. `gpio_read()` requires: access to global (because you forgot to prepend with `local`) function, call.

It will be (apriori) even faster if you cache `gpio.read` to a _local_ variable, as access to local variables is the fastest. So try to use `local gpio_read, gpio_write = gpio.read, gpio.write`. That is what I meant by this and the next  comment.